portal/session.py

```diff
@@ -0,0 +1,84 @@
+import os
+
+from flask_session import Session, RedisSessionInterface
+from flask_session.sessions import total_seconds
+
+
+class BaseRedisSessionInterface(RedisSessionInterface):
+    """Override RedisSessionInterface save_session() method to passthrough flask config"""
+
+    def save_session(self, app, session, response):
+        domain = self.get_cookie_domain(app)
+        path = self.get_cookie_path(app)
+        if not session:
```

Comment:
is this `not` a typo?  w/i a negated test for session, you check if it's modified?

Reply:
I'll preface that most of the code in these overridden methods is copied directly from the parent class they extend: [`RedisSessionInterface`](https://github.com/fengsp/flask-session/blob/0.3.2/flask_session/sessions.py#L90-L173)

If I understand correctly, `session.modified` tracks if the incoming session data differs from the existing session data.
This nested condition handles the case where the session data has been cleared, but hadn't been reflected in the server-side persistent storage (redis) yet.